broadinstitute/single_cell_portal_core
# Adding mtx file testing

Testing of happy path of mtx file upload.  Creates a new upload-wizard-test-utils.js file to consolidate common test patterns.  

Also fixes a UX papercut where the message "A file named already exists in your study" would appear if you hovered over the barcodes file save button before selecting a file.  This was a result of the features and barcodes file objects being created, both with a name of '', and therefore duplicating each other.  

TO TEST:
1. Run 'yarn test' -- confirm everything passes
2. in the new upload wizard, select 'Sparse Matrix (.mtx)'
3. Hover over the 'Save' buttons for the features and barcodes files.  Confirm two messages appear: 'You must save the parent file' and 'You must select a file'. but NOT 'a file named already exists'
